jax/lax/lax_parallel.py

```diff
@@ -531,11 +546,23 @@ def _moveaxis(src, dst, x):
   perm.insert(dst, src)
   return lax.transpose(x, perm)
 
+def _all_to_all_batcher(vals_in, dims_in, *, axis_name, split_axis, concat_axis):
+  x, = vals_in
+  d, = dims_in
+  if split_axis <= d:
+    split_axis += 1
+  if concat_axis <= d:
+    concat_axis += 1
+  result = all_to_all_p.bind(x, axis_name=axis_name, split_axis=split_axis, concat_axis=concat_axis)
+  return result, d
```

Comment:
Hm I just realized that I might need to adjust `d` here...

Reply:
Maybe try to add a test that fails?

Reply:
I think it should be 

```python
# Vmapped axis got shifted down one.
if split_axis < concat_axis: 
  if split_axis <= d and concat_axis >= d:
    d -= 1
# Vmapped axis got shifted up one. 
if split_axis > concat_axis:
  if split_axis >= d and concat_axis <= d:
    d += 1
# In the case where the are equal, leave d alone.
```